src/EA.Weee.Database/scripts/Update/Release 7/Sprint4/20241003-1016-producer-eee-sp-update.sql

```diff
@@ -0,0 +1,420 @@
+--		EEEData.Cat14 as Cat14Q4
+--	 from #ProducerEEEData EEEData
+--	 where EEEData.Quarter = 4
+--) as Q4 on (Producers.PRN= Q4.PRN AND Producers.SchemeId = Q4.SchemeId)
+
+--Order by SchemeName, ProducerName
+--END
+
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+
+ALTER PROCEDURE [Producer].[spgProducerEeeCsvData]
+    @ComplianceYear INT,
+    @SchemeId UNIQUEIDENTIFIER = NULL,
+    @ObligationType NVARCHAR(4)
+AS
+BEGIN
+    -- Create temporary table to store EEE data
+    CREATE TABLE #EEETable (
+        PRN NVARCHAR(100),
+        SchemeId UNIQUEIDENTIFIER,
+        ApprovalNumber NVARCHAR(50),
+        SchemeName NVARCHAR(255),
+        ProducerName NVARCHAR(255),
+        ProducerCountry NVARCHAR(100),
+        Quarter INT,
+        WeeeCategory INT,
+        Tonnage DECIMAL(18, 3),
+        IsDirectProducer BIT
+    )
+
+    -- Insert data into temporary table
+    INSERT INTO #EEETable
+    SELECT 
+        RP.ProducerRegistrationNumber AS 'PRN',
+        RP.SchemeId,
+        S.ApprovalNumber,
+        S.SchemeName,
+        COALESCE(PBC.Name, PBP.Name, '') AS 'ProducerName',
+        COALESCE(ROC_A_C.Name, PPOB_A_C.Name, '') AS 'ProducerCountry',
+        DR.Quarter,
+        EEOA.WeeeCategory,
+        EEOA.Tonnage,
+        0 AS IsDirectProducer  -- Not a direct producer
+    FROM [PCS].DataReturn DR
+    INNER JOIN [PCS].DataReturnVersion DRV ON DR.CurrentDataReturnVersionId = DRV.Id
+    INNER JOIN [PCS].[EeeOutputReturnVersion] EEORV ON DRV.EeeOutputReturnVersionId = EEORV.Id
+    INNER JOIN [PCS].[EeeOutputReturnVersionAmount] EEORVA ON EEORV.Id = EEORVA.EeeOutputReturnVersionId
+    INNER JOIN [PCS].[EeeOutputAmount] EEOA ON EEORVA.EeeOuputAmountId = EEOA.Id
+    INNER JOIN [Producer].[RegisteredProducer] RP ON EEOA.RegisteredProducerId = RP.Id
+    INNER JOIN [PCS].Scheme S ON RP.SchemeId = S.Id
+    INNER JOIN [Producer].[ProducerSubmission] PS ON RP.[CurrentSubmissionId] = PS.[Id]
+    LEFT JOIN [Producer].[Business] PB ON PS.ProducerBusinessId = PB.Id
+    LEFT JOIN [Producer].[Company] PBC ON PB.CompanyId = PBC.Id
+    LEFT JOIN [Producer].[Partnership] PBP ON PB.PartnershipId = PBP.Id
+    LEFT JOIN [Producer].[Contact] ROC 
+        INNER JOIN [Producer].[Address] ROC_A 
+            INNER JOIN [Lookup].[Country] ROC_A_C ON ROC_A.CountryId = ROC_A_C.Id     
+        ON ROC.AddressId = ROC_A.Id
+    ON PBC.RegisteredOfficeContactId = ROC.Id
+    LEFT JOIN [Producer].[Contact] PPOB 
+        INNER JOIN [Producer].[Address] PPOB_A 
+            INNER JOIN [Lookup].[Country] PPOB_A_C ON PPOB_A.CountryId = PPOB_A_C.Id   
+        ON PPOB.AddressId = PPOB_A.Id
+    ON PBP.PrincipalPlaceOfBusinessId = PPOB.Id
+    WHERE DR.ComplianceYear = @ComplianceYear
+        AND RP.Removed = 0
+        AND EEOA.ObligationType = @ObligationType
+        AND (@SchemeId IS NULL OR S.[Id] = @SchemeId)
+
+    UNION ALL
+
+    SELECT
+        RP.ProducerRegistrationNumber as 'PRN',
+        dr.Id AS 'SchemeId',
```

Comment:
The only real change to this stored procedure is here to include the new direct producer submissions, the rest was tidy up